### ISSUE TITLE
Result of ByteBufUtil.compare(ByteBuf a, ByteBuf b) is dependent on ByteOrder of supplied ByteBufs

### ISSUE THREAD

**Issue by adityakishore** [netty/netty#5718]
This should be similar to memcmp() or strcmp() as described in the java-doc.

This is because the [ByteBufUtil.compare(ByteBuf, ByteBuf)](https://github.com/netty/netty/blob/4.1/buffer/src/main/java/io/netty/buffer/ByteBufUtil.java#L226) compare the `ByteBuf`s as integers and not raw bytes.

```
Sample Code
========
  public static void main(String[] args) {
    ByteBuf buf1 = Unpooled.wrappedBuffer("1234".getBytes()).order(ByteOrder.LITTLE_ENDIAN);
    ByteBuf buf2 = Unpooled.wrappedBuffer("4321".getBytes()).order(ByteOrder.LITTLE_ENDIAN);
    System.out.println("\"1234\".compareTo(\"4321\") = " + buf1.compareTo(buf2));

    buf1 = Unpooled.wrappedBuffer("1234".getBytes()).order(ByteOrder.BIG_ENDIAN);
    buf2 = Unpooled.wrappedBuffer("4321".getBytes()).order(ByteOrder.BIG_ENDIAN);
    System.out.println("\"1234\".compareTo(\"4321\") = " + buf1.compareTo(buf2));
  }

Output
=====
"1234".compareTo("4321") = 1
"1234".compareTo("4321") = -1
```

**Comment by normanmaurer** [comment]
@adityakishore so you basically say both should return 1. Right ?

**Comment by normanmaurer** [comment]
@adityakishore ups -1 I mean :)

**Comment by normanmaurer** [comment]
Let me fix this.

**Comment by adityakishore** [comment]
Yes, both should return -1 if we want to retain the memcmp/strcmp behavior.

For that, both buffer should be compared as BIG_ENDIAN if we still want to retain the performance optimization of comparing 4 bytes at a time.

**Comment by normanmaurer** [comment]
@adityakishore PTAL https://github.com/netty/netty/pull/5732

**Comment by Scottmitch** [comment]
fixed by #5732

**Comment by adityakishore** [comment]
Thanks for fixing this.
